src_kt/java/arcs/crdt/entity/FieldValue.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.crdt.entity
+
+import arcs.crdt.internal.Referencable
+import arcs.crdt.internal.ReferenceId
+
+/**
+ * Referencable wrapper of an actual value for an entity's field.
+ *
+ * **Note:** A [FieldValueInterpreter] must have been registered for [T] before instantiation.
+ */
+data class FieldValue<T : Any>(
+  /** [ReferenceId] for the value. */
+  override val id: ReferenceId,
+  /** Raw string-representation of the value. */
+  val serializedValue: String,
+  /** @deprecated */
+  @Deprecated(
+    "Use getValue() instead.",
+    ReplaceWith("getValue()"),
+    DeprecationLevel.WARNING
+  )
+  var actualValue: T? = null
```

Comment:
I'm so confused. How can brand new code already be deprecated???

Reply:
It's not that it's actually deprecated per-se, however if I want to support clean [de]serialization using `inline fun <reified T>`, all of the fields that the inline functions access need to have the same visibility as the functions.  I wish we had a `@Hide` annotation available, like Android does for their codebase - to hide things that need to be public for the framework, but shouldn't be used by clients.